scipy/optimize/_minimize.py

```diff
@@ -581,50 +582,79 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
+        res = _minimize_lbfgsb(fun, x0, args, jac, bounds,
                                 callback=callback, **options)
     elif meth == 'tnc':
-        return _minimize_tnc(fun, x0, args, jac, bounds, callback=callback,
+        res = _minimize_tnc(fun, x0, args, jac, bounds, callback=callback,
                              **options)
     elif meth == 'cobyla':
-        return _minimize_cobyla(fun, x0, args, constraints, **options)
+        res = _minimize_cobyla(fun, x0, args, constraints, **options)
     elif meth == 'slsqp':
-        return _minimize_slsqp(fun, x0, args, jac, bounds,
+        res = _minimize_slsqp(fun, x0, args, jac, bounds,
                                constraints, callback=callback, **options)
     elif meth == 'trust-constr':
-        return _minimize_trustregion_constr(fun, x0, args, jac, hess, hessp,
+        res = _minimize_trustregion_constr(fun, x0, args, jac, hess, hessp,
                                             bounds, constraints,
                                             callback=callback, **options)
     elif meth == 'dogleg':
-        return _minimize_dogleg(fun, x0, args, jac, hess,
+        res = _minimize_dogleg(fun, x0, args, jac, hess,
                                 callback=callback, **options)
     elif meth == 'trust-ncg':
-        return _minimize_trust_ncg(fun, x0, args, jac, hess, hessp,
+        res = _minimize_trust_ncg(fun, x0, args, jac, hess, hessp,
                                    callback=callback, **options)
     elif meth == 'trust-krylov':
-        return _minimize_trust_krylov(fun, x0, args, jac, hess, hessp,
+        res = _minimize_trust_krylov(fun, x0, args, jac, hess, hessp,
                                       callback=callback, **options)
     elif meth == 'trust-exact':
-        return _minimize_trustregion_exact(fun, x0, args, jac, hess,
+        res = _minimize_trustregion_exact(fun, x0, args, jac, hess,
                                            callback=callback, **options)
     else:
         raise ValueError('Unknown solver %s' % method)
 
+    # cannot be a successful minimisation if the bounds are exceeded, or
+    # constraints aren't met
+    if bounds is not None:
+        # force to be new-style
+        b = standardize_bounds(bounds, res.x, 'trust-constr')
+        if np.any(res.x > b.ub) or np.any(res.x < b.lb):
+            res.success = False
+
+    if constraints is not None and len(constraints):
+        catol = 0.
+        if meth == 'cobyla':
+            catol = 2e-4
+            if 'catol' in options:
+                catol = options['catol']
+
+        x = np.squeeze(res.x)
+        # force to be new-style
+        cons = standardize_constraints(constraints, res.x, 'trust-constr')
+        pcons = [PreparedConstraint(con, x) for con in cons]
+        excesses = np.concatenate([pcon.violation(x) for
+                                   pcon in pcons])
+        if np.any(excesses > catol):
+            res.success = False
```

Comment:
It would be good to add a status message that corresponds with algorithm terminating successfully but not satisfying constraints.

Reply:
In `linprog`:
```The solution does not satisfy the constraints within the required tolerance of 3.16E-04, yet no errors were raised and there is no certificate of infeasibility or unboundedness.```
...
```Check whether the slack and constraint residuals are acceptable; if not, consider enabling presolve, reducing option `tol`, and/or using method revised simplex. If you encounter this message under different circumstances, please submit a bug report.```

Reply:
+1 for that